spec/lib/reports/sales_tax_totals_by_order_spec.rb

```diff
@@ -0,0 +1,258 @@
+      [
+        [state_tax_rate.id, order.id],
+        order
+      ]
+    end
+
+    it "returns the tax amount filtered by tax rate in the query_row" do
+      OrderWorkflow.new(order).complete!
+
+      tax_total = report.tax_rate_total(query_row)
+
+      expect(tax_total).not_to eq(order.total_tax)
+
+      # 20 % of 10.00 shipment cost + 20 % of 100.00 line item price
+      expect(tax_total).to eq(0.2 + 2)
+    end
+
+    context "with a voucher" do
+      let(:voucher) do
+        create(:voucher_flat_rate, code: 'some_code', enterprise: order.distributor, amount: 10)
+      end
+
+      it "returns the tax amount adjusted with voucher tax discount" do
+        add_voucher(order, voucher)
+
+        tax_total = report.tax_rate_total(query_row)
+
+        # 20 % of 10.00 shipment cost + 20 % of 100.00 line item price - voucher tax
+        expect(tax_total).to eq(0.2 + 2 - 0.29)
+      end
+    end
+  end
+
+  describe "#total_excl_tax" do
+    let(:query_row) do
+      [
+        [state_tax_rate.id, order.id],
+        order
+      ]
+    end
+
+    it "returns the total excluding tax specified in query_row" do
+      OrderWorkflow.new(order).complete!
+
+      total = report.total_excl_tax(query_row)
+
+      # order.total - order.total_tax
+      expect(total).to eq(113.3 - 3.3)
+    end
+
+    context "with a voucher" do
+      let(:voucher) do
+        create(:voucher_flat_rate, code: 'some_code', enterprise: order.distributor, amount: 10)
+      end
+
+      it "returns the total exluding tax and indcluding voucher tax discount" do
+        add_voucher(order, voucher)
+
+        total = report.total_excl_tax(query_row)
+
+        # order.total - order.total_tax + voucher tax
+        expect(total).to eq(103.3 - 3.3 + 0.29)
+      end
+    end
+  end
+
+  describe "#total_incl_tax" do
+    let(:query_row) do
+      [
+        [state_tax_rate.id, order.id], order
+      ]
+    end
+
+    it "returns the total including the tax specified in query_row" do
+      OrderWorkflow.new(order).complete!
+
+      total = report.total_incl_tax(query_row)
+
+      # order.total - order.total_tax + filtered tax
+      expect(total).to eq(113.3 - 3.3 + 2.2)
+    end
+  end
+
+  describe "#rules" do
+    before do
+      OrderWorkflow.new(order).complete!
+    end
+
+    it "returns rules" do
+      expected_rules = [
+        {
+          group_by: :distributor,
+        },
+        {
+          group_by: :order_cycle,
+        },
+        {
+          group_by: :order_number,
+          summary_row: an_instance_of(Proc)
+        }
+      ]
+
+      expect(report.rules).to match(expected_rules)
+    end
+
+    describe "summary_row" do
+      it "returns expected totals" do
+        expected_summary = {
+          total_excl_tax: 110.00,
+          tax: 3.3,
+          total_incl_tax: 113.30,
+          first_name: "cfname",
+          last_name: "clname",
+          code: "ABC123",
+          email: "order1@example.com"
+        }
```

Comment:
I would embed the expected values down below in the `expect` statement at the end, I think it reads better that way.